### PR TITLE
Madlib 457

### PR DESCRIPTION
**Pull request by renyi533**
Fix madlib-457. 

When comparing the C string str1 to the C string str2, we interpret them appropiately according to the LC_COLLATE category of the current locale.

### CODE REVIEW

**Comment by cwelton** [comment]
The change looks reasonable.
- It would be good to add a test case that demonstrates the fix.
- I would encourage spending some time learning a bit more about git.  There is no reason this pull request needs 11 commits associated with it, the single commit that fixes the issue is all that is needed.  E.g. commit b500b91 "cart does not rely on svec and array_op now. Remove the dependency" is not related to this change, it should not be in the pull request.

**Comment by renyi533** [comment]
I will add test cases for it.

Yes. I am still not familiar with git. I will learn more. For future PR, I will make the commit list more reasonable.

**Comment by renyi533** [comment]
I added the test case for it. And I verified that the test case can pass. Please check the result of "svec|gp_sfv_sort_order.sql_in" below.

madpack.py : INFO : Detected Greenplum DB version 4.1.
TEST CASE RESULT|Module: array_ops|array_ops.sql_in|PASS|Time: 243 milliseconds
TEST CASE RESULT|Module: bayes|bayes.sql_in|PASS|Time: 3812 milliseconds
TEST CASE RESULT|Module: compatibility|gp_comp.sql_in|PASS|Time: 1683 milliseconds
TEST CASE RESULT|Module: cart|dt_test.sql_in|PASS|Time: 318397 milliseconds
TEST CASE RESULT|Module: cart|dt_utility_test.sql_in|PASS|Time: 689 milliseconds
TEST CASE RESULT|Module: plda|plda.sql_in|PASS|Time: 13482 milliseconds
TEST CASE RESULT|Module: prob|prob.sql_in|PASS|Time: 65 milliseconds
TEST CASE RESULT|Module: quantile|quantile.sql_in|PASS|Time: 375 milliseconds
TEST CASE RESULT|Module: sketch|cm.sql_in|PASS|Time: 15198 milliseconds
TEST CASE RESULT|Module: sketch|fm.sql_in|PASS|Time: 1805 milliseconds
TEST CASE RESULT|Module: sketch|mfv.sql_in|PASS|Time: 335 milliseconds
TEST CASE RESULT|Module: sketch|support.sql_in|PASS|Time: 63 milliseconds
TEST CASE RESULT|Module: svd_mf|svdmf.sql_in|PASS|Time: 13745 milliseconds
TEST CASE RESULT|Module: svec|gp_sfv_sort_order.sql_in|PASS|Time: 495 milliseconds
TEST CASE RESULT|Module: svec|svec_test.sql_in|PASS|Time: 3271 milliseconds
TEST CASE RESULT|Module: textfex_viterbi|textfex_viterbi.sql_in|PASS|Time: 2227 milliseconds
TEST CASE RESULT|Module: assoc_rules|assoc_rules.sql_in|PASS|Time: 3554 milliseconds
TEST CASE RESULT|Module: conjugate_gradient|conj_grad.sql_in|PASS|Time: 878 milliseconds
TEST CASE RESULT|Module: data_profile|data_profile.sql_in|PASS|Time: 1862 milliseconds
TEST CASE RESULT|Module: kmeans|kmeans.sql_in|PASS|Time: 19122 milliseconds
TEST CASE RESULT|Module: kernel_machines|kernel_mach.sql_in|PASS|Time: 9237 milliseconds
TEST CASE RESULT|Module: regress|linear.sql_in|PASS|Time: 1203 milliseconds
TEST CASE RESULT|Module: regress|logistic.sql_in|PASS|Time: 3597 milliseconds

File: methods/svec/src/pg_gp/sql/gp_sfv_sort_order.sql_in [hunk 1]
```diff
@@ -0,0 +1,56 @@
+CREATE TEMP TABLE dictionary (
+    rank bigint,
+    term text,
+    count bigint
+) m4_ifdef(`GREENPLUM',`DISTRIBUTED BY (rank)');
+
+
+COPY dictionary (rank, term, count) FROM stdin;
+1	̮	1
+2
```
Comment (haradh1) [inline methods/svec/src/pg_gp/sql/gp_sfv_sort_order.sql_in:10]: Don't use madlib schema directly in test SQL.

**Comment by haradh1** [comment]
The added test case failed in my PG 9.1.  It passed in PG 9.0, so you need to look at the collation support in 9.1.

``` sql
select
        madlib.svec_sfv( dict.dict, ARRAY['0'] ) sfv
    from
        ( select dict from dictionary_array limit 1 ) dict
;
psql:/tmp/madlib.nC9HGy/svec/test/gp_sfv_sort_order.sql_in.tmp:62: ERROR:  Dictionary is unsorted: '<U+1F380><U+1F49D>' is out of order.
```

**Comment by haradh1** [comment]
The failure was my environmental issue.  Fixing it, the test now passes.  Sorry for noise.

**Comment by haradh1** [comment]
Pulled.

**Pull request closed without merging**